diff --git '${BUILD_DIR}'uboot-linux4sam_5.7/common/env_common.c '${BUILD_DIR}'uboot-linux4sam_5.7/common/env_common.c
index 7fb62e8b45..a1cf18a973 100644
--- '${BUILD_DIR}'uboot-linux4sam_5.7/common/env_common.c
+++ '${BUILD_DIR}'uboot-linux4sam_5.7/common/env_common.c
@@ -124,6 +124,7 @@ void set_default_env(const char *s)
 
 	gd->flags |= GD_FLG_ENV_READY;
 	gd->flags |= GD_FLG_ENV_DEFAULT;
+	saveenv();
 }
 
 
